src/pilot/assign_role.py

```diff
@@ -428,9 +459,11 @@ def define_storage_logical_disks(drac_client, raid_controller_name):
     #
     # A successful call returns a list, which may be empty; otherwise,
     # None is returned.
-    jbod_capable = drac_client.is_jbod_capable(raid_controller_name)
+    if not raid_controller:
+        raid_controller = boss_controller
+    jbod_capable = drac_client.is_jbod_capable(raid_controller)
     jbod_logical_disks = define_jbod_logical_disks(
-        drac_client, remaining_physical_disks, raid_controller_name,
+        drac_client, remaining_physical_disks, raid_controller,
         jbod_capable)
 
     if jbod_logical_disks is None:
```

Comment:
This "if" statement doesn't look right to me.  It should be perfectly ok to define an os_logical_disk and no jbod_logical_disks.  This is what would happen in the PERC H740P case, and in the HBA330 case too.  I think lines 469-470 should be deleted.